mina-staking-ledgers-exporter/scripts/mina-staking-ledgers-exporter.py

```diff
@@ -40,7 +40,10 @@ def main():
     logger.info(f" -- Processing current epoch ({current_epoch}) --")
     process_staking_ledger(synced_pod, current_epoch, "staking-epoch-ledger")
     logger.info(f" -- Processing next epoch ({next_epoch}) --")
-    process_staking_ledger(synced_pod, next_epoch, "next-epoch-ledger")
+    try:
+      process_staking_ledger(synced_pod, next_epoch, "next-epoch-ledger")
+    except Exception as e:
+      logger.warning(f"Unable to process the next epoch {next_epoch} due to error: {e}, skipping further processing.")
```

Comment:
**suggestion:** I suggest we catch only when the error message is `Ledger not found: next staking ledger is not finalized yet`.

```suggestion
    except subprocess.CalledProcessError as e:
        if "Ledger not found: next staking ledger is not finalized yet" in e.output:
            logger.info("Ledger not finalized yet, ignore for now")
        else:
            raise e
```

Reply:
Nice catch. thanks @johnmarcou